### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by samirelanduk** [molstar/molstar#124]
First let me say that from what I've used of molstar I think it's really great - a big improvement on NGL which was already superb.

That said, I'm worried at the lack of any usable documentation. I read #81, which was closed without really addressing the main point - without detailed documentation, or at least something analogous to the NGL docs, this library is very, very difficult to use. This would be a shame for any library, but for one as powerful as this, it's such a waste.

I know you all do a lot with a small team, and you have to prioritise what you spend your time on, but I would urge to reconsider the importance of at least a quick start guide and overview of the core concepts/object types. Examples are not really documentation, unless you want to use the specific features illustrated in those examples. For example I am currently struggling with how to do something as simple as changing the background colour of the viewer, which I'm sure must be possible.

In any case, thanks for creating one of the coolest libraries in structural biology.

**Comment by arose** [comment]
Thank you Sam! I very much agree that documentation is needed to flatten the learning curve and make molstar accessible. I would be very happy work with anyone who wants start a developer documentation for molstar, maybe starting with the viewer app and the plugin context as a suggestion.

**Comment by samirelanduk** [comment]
That's great, thanks. I would be happy to help, but it's a sort of Catch 22 where you need documentation to be able to understand the library well enough to write documentation. That said I have been looking over the code for a while with a view to moving from NGL to this in the long term, so I might be able to get a good enough handle on it to start on something over the next few months. I'll let you know before starting anything major to check that it's along the right lines, if that's alright.

**Comment by dsehnal** [comment]
Glad you are willing to help! Here are some ideas:

- Pick good framework for the documentation / suggest a couple ones and we choose one together with you and Alex
- Create a skeleton of the documentation with topics that should be covered and leave the actual text/code example empty (where you are unsure) and me and Alex could gradually fill that up.
- Maybe we should create the documentation as a separate project within the Mol* organazation where more people could contribute over time.

What do you think?

**Comment by dsehnal** [comment]
I set up https://github.com/molstar/docs with MkDocs which is then available at https://molstar.org/docs/.

Any contributions to the documentation are welcome!

**Comment by samirelanduk** [comment]
This is fantastic, thanks. Once my thesis is submitted in a month or so I'd really like to contribute to this if possible.

**Comment by rtviii** [comment]
Hey, @dsehnal. I've been using MOLE for the past year and have been eyeing Molstar for at least as much, am finally getting to integrating it into our ribosomal database frontend. Really grateful for your work. 

I'm unfamilliar with the code as of today but have indeed wished for better documentation already while trying to find my way around here. I'd be very willing to help write the doc for the viewer as i'm integrating it into my react app. Any suggestions on where to start are much appreciated, likewise if you wanted to assign something to me.

**Comment by arose** [comment]
HI @rtviii, for integrating Mol* into a react app look here https://molstar.org/docs/plugin/. Hopefully that helps and if you have any suggestions on how to improve it let us know or even better modify the docs themselves here https://github.com/molstar/docs. Some open topics are here https://molstar.org/docs/data-state/ if you would like to fill in some blanks.

**Comment by arose** [comment]
superseded by #138
